xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapGroupProvider.java

```diff
@@ -252,13 +321,24 @@ public int getGroupCount() {
         filter.append(MessageFormat.format(manager.getGroupSearchFilter(), "*"));
         filter.append('(').append(key).append('=').append(LdapManager.sanitizeSearchFilter(value, false));
         filter.append("))");
+        
+        String ldapfilter = filter.toString();
+        String searchRangeStr = ";range=";
+        if (ldapfilter.contains(searchRangeStr))
+        {
+            ldapfilter=ldapfilter.substring(0,ldapfilter.indexOf(searchRangeStr))+
+                    ldapfilter.substring(ldapfilter.indexOf("=", 
+                            ldapfilter.indexOf(searchRangeStr)+searchRangeStr.length()));
+        }
+        
         if (Log.isDebugEnabled()) {
-            Log.debug("Trying to find group names using query: " + filter.toString());
+            Log.debug("Trying to find group names using query: " + ldapfilter);
```

Comment:
While you're at it, you might as well use parameters:

    Log.debug("Trying to find group names using query: {}", ldapfilter);

With that, I don't think that there's a point in keeping the `if (Log.isDebugEnabled())` check.

Reply:
done...